### DISCUSSION TITLE
Emails & Messaging app returns "Missing auth data" on save

### DISCUSSION THREAD

**Discussion by falko-burghausen** [saleor/apps#1080] [Q&A]
Hi,

I installed the emails & messaging app in a self-hosted Saleor setup, the app is running and I am able to start with the configuration. However, as soon as I try to save the SMTP configuration, I get an error message "Missing auth data". The authorization bearer token is set in the header in this request, also the Saleor API URL header is set to the correct value. Saleor Dashboard and core are both running on v3.16.1. The SMTP credentials are definitively correct, as I am already using them with the deprecated email plugin, although I think this error has nothing to do with the SMPT credentials. 

<img width="1710" alt="Bildschirmfoto 2023-10-24 um 16 57 38" src="https://github.com/saleor/apps/assets/22614664/a199e5fc-2183-43b4-8de1-1137852f5df2">

Any idea what could cause this issue and how to resolve it?

Thanks!

**Comment by witoszekdev** [comment]
Hi @falko-burghausen, are you also self-hosting the app?

Installing our deployed apps (in this case [emails-and-messages.saleor.app](https://emails-and-messages.saleor.app/)) is only possible in Saleor Cloud.

If you'd like to use the app on a self-hosted Saleor you will also need to self-host the Emails & Messages app (for example on Vercel). To do that, please follow this guide: https://docs.saleor.io/docs/3.x/developer/extending/apps/developing-apps/app-examples#saleor-apps

After you've forked the repo, you'll need to configure the environment variables for the app: https://docs.saleor.io/docs/3.x/developer/extending/apps/developing-apps/app-examples#environment-variables. Specifically you'd need to set up an APL. For production, we suggest using Upstash APL: https://docs.saleor.io/docs/3.x/developer/extending/apps/developing-apps/app-sdk/apl#upstashapl

**Reply by falko-burghausen** [reply]
Hi @witoszekdev , thanks for your help. Yes, I am self-hosting the app (dockerized at AWS). The installation process works fine up to the point where I want to configre the SMTP provider. 

I've now set the following environment variables at build time (they're are applied during `pnpm build` of the app) for:
a) `SECRET_KEY`
b) `UPSTASH_REDIS_REST_TOKEN`, `UPSTASH_REDIS_REST_URL` and `APL=upstash` (I also tried with `UPSTASH_TOKEN` and `UPSTASH_URL`)

Unfortunately it still does not make any difference, the error message when saving the SMTP provider is the same (_Could not save the configuration, Missing auth data_). Any other idea what's going on here?
